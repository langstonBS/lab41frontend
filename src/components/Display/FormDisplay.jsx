import React, { useState } from 'react'
import { postCereal} from '../../service/cerealService'


function FormDisplay() {

  const [cerealName, setCerealName] = useState('');
  const [cerealThoughs, setCerealThoughs] = useState('');
  const [cerealRatings, setCerealRatings] = useState('');

  const handleChange = ({target}) => {
    if(target.cerealName === 'cerealName') setCerealName(target.value);
    if(target.cerealThoughs === 'cerealThoughs') setCerealThoughs(target.value);
    if(target.cerealRatings === 'cerealRatings') setCerealRatings(target.cerealRatings)
  }

  const handleSubmit = event => {
    event.preventDefault();
    postCereal(cerealName,cerealThoughs,cerealRatings)
  }

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <label>
          <h2>add a Cereal and rate it</h2>
        </label>
        <input
          input type="text"
          value={cerealName}
          onChange={handleChange} />
        <label>
          <h4>Talk about it</h4>
        </label>
        <textarea
          type='text'
          value={cerealThoughs}
          onChange={handleChange} >
          </textarea>
        <label>
          <h4>Rate it</h4>
        </label>
        <input type='number'
          value={cerealRatings}
          onChange={handleChange} />
        <br></br>
        <br></br>
        <button>Submit</button>
      </form>
    </div>
  )
}

export default FormDisplay
