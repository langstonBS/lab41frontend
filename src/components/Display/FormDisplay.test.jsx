import React from 'react';
import { shallow } from 'enzyme';
import FormDisplay from './FormDisplay';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<FormDisplay />);
    expect(wrapper).toMatchSnapshot();
  });
});