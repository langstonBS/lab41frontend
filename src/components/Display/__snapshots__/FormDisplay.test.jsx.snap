// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
      <h2>
        add a Cereal and rate it
      </h2>
    </label>
    <input
      input={true}
      onChange={[Function]}
      type="text"
      value=""
    />
    <label>
      <h4>
        Talk about it
      </h4>
    </label>
    <textarea
      onChange={[Function]}
      type="text"
      value=""
    />
    <label>
      <h4>
        Rate it
      </h4>
    </label>
    <input
      onChange={[Function]}
      type="number"
      value=""
    />
    <br />
    <br />
    <button>
      Submit
    </button>
  </form>
</div>
`;
